refactor(createRoutes): leverage default exports

devopsmariocom opened the pull request:
to simplify module loading logic

devopsmariocom committed:
refactor(createRoutes): leverage default exports

to simplify module loading logic

Hotell commented on src/routes.tsx:
```diff
@@ -12,12 +12,8 @@ const loadModule = ( cb ) => ( componentModule ) => {
 };
```
cb needs annotation:

```
cb: (err:any, component: Component<any,any>)=>void
```

https://github.com/reactjs/react-router/blob/master/docs/API.md#getcomponentnextstate-callback

or check it in type definitions if there isn't a type for that already

Hotell replied:
hmm I don't see a point here why loadModule is a closure, when it can be loaded via one function call, see docs

Hotell commented:
pls read this before doing anything https://github.com/reactjs/react-router/blob/master/docs/guides/DynamicRouting.md

Hotell commented:
I'm not very comfortable to export things both by default and as named, but maybe for this particular reasons it makes sense.

Also the thing is that connected components should be exported by default and named for "testing" purposes

devopsmariocom committed:
refactor(createRoutes): set any for incompatible and types prevent implicit any

type definitions seems tobe off by a lot, so let's go with any before we or somebody else will fix the types

devopsmariocom commented:
read https://github.com/reactjs/react-router/blob/master/docs/guides/DynamicRouting.md anything specific you want to point out?

devopsmariocom commented:
ok so let's go with default exports for pages then same as for linked components
